### PR TITLE
Feature/144 -  skill calculation

### PR DESCRIPTION
**Pull request by SaXaPhonist**
- add skill calculation by course complexity

### CODE REVIEW

**Review by k-krasilnikova — APPROVED** [review]

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]
Remove pelase using of the complexity in calc

File: backend/src/utils/dto/skillsDto.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 import { containsObjectId } from 'utils/comparator/ObjectId/containsObjectId';
+import { ICourse } from 'interfaces/Ientities/Icourses';
 
-const specifySkills = (userSkills: IUserSkill[], techArrayIds: ObjectId[]) => {
+const specifySkills = (userSkills: IUserSkill[], techArrayIds: ICourse['technologies']) => {
   const skillIds = userSkills.map((userSkill) => userSkill.skill);
-  const newSkills = techArrayIds.filter((techId) => !containsObjectId(skillIds, techId));
-  const oldSkills = techArrayIds.filter((techId) => containsObjectId(skillIds, techId));
+  const newSkills = techArrayIds.filter((techId) => !containsObjectId(skillIds, techId.skill));
+  const oldSkills = techArrayIds.filter((techId) => containsObjectId(skillIds, techId.skill));
```
Comment (dapodev) [inline backend/src/utils/dto/skillsDto.ts:8]: Let's work on naming here. It's not `techArrayIds` and not `techId` anymore

File: backend/src/interfaces/Ientities/Icourses.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { UserRank } from 'enums/users';
 interface ICourse {
   _id?: ObjectId;
   title: string;
-  technologies: ObjectId[];
+  technologies: { skill: ObjectId; points: number }[];
   requiredSkills?: ObjectId[];
```
Comment (dapodev) [inline backend/src/interfaces/Ientities/Icourses.ts:9]: We don't use `requiredSkills` anywhere. If @k-krasilnikova agree, I would propose to remove this field from models, interfaces and migrations.

**Review by dapodev — CHANGES_REQUESTED** [review]

File: backend/src/interfaces/Ientities/Icourses.ts (hunk 2 above)
Comment (k-krasilnikova) [reply backend/src/interfaces/Ientities/Icourses.ts:9]: Yep, let's do it in a separate PR @SaXaPhonist @dapodev

**Pull request merged**